h5py/_hl/selections.py

```diff
@@ -91,6 +91,68 @@ def select(shape, args, dsid):
     return sel
 
 
+class MultiBlockSlice(object):
+
+    """
+        A conceptual extension of the built-in slice object to allow selections
+        using start, stride, count and block.
+
+        If given, these parameters will be passed directly to
+        H5Sselect_hyperslab. The defaults are start=0, stride=1, block=1,
+        count=length, which will select the full extent.
+
+        __init__(start, count, stride, block) => Create a new MultiBlockSlice, storing
+            any given selection parameters and using defaults for the others
+        start => The offset of the starting element of the specified hyperslab
+        count => The number of blocks to select
+        stride => The number of elements between the start of one block and the next
+        block => The number of elements in each block
+
+    """
+
+    def __init__(self, start=0, count=None, stride=1, block=1):
```

Comment:
Is there a reason for this ordering? In [H5Sselect_hyperslab](https://confluence.hdfgroup.org/display/HDF5/H5S_SELECT_HYPERSLAB), the order is start, stride, count, block.

Reply:
I think I just defined it in the same order as `handle_slice`.

Reply:
`handle_slice` isn't public, so there's no particular need to match it. I'd match the order in the HDF5 function unless we can come up with a solid reason to put them in another order.

Reply:
Good point. I will update the `__init__`, but leave it as is elsewhere.

Reply:
Oh, and in `__repr__`